src/rdm/mdb.c

```diff
@@ -0,0 +1,263 @@
+        break;  // Don't encode null terminators
+      }
+    }
+  }
+  mdb->pdl = encoded;
+  return encoded;
+}
+
+size_t rdm_encode_nack_reason(rdm_mdb_t *mdb, rdm_nr_t nack_reason) {
+  return rdm_encode_16bit(mdb, &nack_reason, 1);
+}
+
+size_t rdm_encode_null(rdm_mdb_t *mdb, const void *data, int num) {
+  mdb->pdl = 0;
+  return 0;
+}
+
+int rdm_decode_mute(const rdm_mdb_t *mdb, void *data, int num) {
+  int decoded = 0;
+  if (mdb && mdb->pdl && data) {
+    const struct rdm_disc_mute_data_t *const pd = (void *)mdb->pd;
+    rdm_disc_mute_t *param = data;
+    param->managed_proxy = pd->managed_proxy;
+    param->sub_device = pd->sub_device;
+    param->boot_loader = pd->boot_loader;
+    param->proxied_device = pd->proxied_device;
+    param->binding_uid = RDM_UID_NULL;  // FIXME
+    // param->binding_uid = mdb->pdl > 2 ? bswap48(pd->binding_uid) : 0;
+    decoded = 1;
+  }
+  return decoded;
+}
+
+size_t rdm_encode_mute(rdm_mdb_t *mdb, const void *data, int num) {
+  size_t encoded = 0;
+  if (mdb && data && num) {
+    struct rdm_disc_mute_data_t *pd = (void *)mdb->pd;
+    const rdm_disc_mute_t *param = data;
+    pd->control_field = 0;  // Zero out bits 4 through 15
+    pd->managed_proxy = param->managed_proxy;
+    pd->sub_device = param->sub_device;
+    pd->boot_loader = param->boot_loader;
+    pd->proxied_device = param->proxied_device;
+    if (!uid_is_null(&param->binding_uid)) {      
+      uidcpy(pd->binding_uid, &(param->binding_uid));
+      encoded += 6;
+    }
+    encoded += 2;
+  }
+  mdb->pdl = encoded;
+  return encoded;
+}
+
+int rdm_decode_device_info(const rdm_mdb_t *mdb, void *data, int num) {
+  int decoded = 0;
+  if (mdb != NULL && data != NULL) {
+    const rdm_device_info_data_t *pd = (void *)mdb->pd;
+    rdm_device_info_t *const restrict param = data;
+    param->model_id = bswap16(pd->model_id);
+    param->product_category = bswap16(pd->product_category);
+    param->software_version_id = bswap32(pd->software_version_id);
+    param->footprint = bswap16(pd->footprint);
+    param->current_personality = pd->current_personality;
+    param->personality_count = pd->personality_count;
+    param->start_address =
+        pd->start_address != 0xffff ? bswap16(pd->start_address) : -1;
+    param->sub_device_count = bswap16(pd->sub_device_count);
+    param->sensor_count = pd->sensor_count;
+    decoded = 1;
+  }
+  return decoded;
+}
+
+size_t rdm_encode_device_info(rdm_mdb_t *mdb, const void *data, int num) {
+  size_t encoded = 0;
+  if (mdb && data) {
+    rdm_device_info_data_t *const pd = (void *)mdb->pd;
+    const rdm_device_info_t *param = data;
+    pd->major_rdm_version = 1;
+    pd->minor_rdm_version = 0;
+    pd->model_id = bswap16(param->model_id);
+    pd->product_category = bswap16(param->product_category);
+    pd->software_version_id = bswap32(param->software_version_id);
+    pd->footprint = bswap16(param->footprint);
+    pd->current_personality = param->current_personality;
```

Comment:
`pd->personality_count`  is missing here.

Reply:
Good catch here as well. I've added the missing line. 

That said, this function will be removed soon and replaced with `rdm_encode()` in `rdm/utils.h`.